gusto/state.py

```diff
@@ -345,15 +351,12 @@ def dump(self, t=0, pickup=False):
             self.pointdata_output.dump(self.fields, t)
 
             # Open the checkpointing file (backup version)
-            files = ["chkptbk", "chkpt"]
-            for file in files:
-                chkfile = path.join(self.dumpdir, file)
-                with DumbCheckpoint(chkfile, mode=FILE_CREATE) as chk:
-                    # Dump all the fields to a checkpoint
-                    for field in self.to_pickup:
-                        chk.store(field)
-                    chk.write_attribute("/", "time", t)
-
+            for file in self.chkpts:
+                # Dump all the fields to a checkpoint
+                for field in self.to_pickup:
+                    file.store(field)
+                file.write_attribute("/", "time", t)
```

Comment:
Prefer not to use builtin names (file).  Write `for chk in self.chkpts`?

Reply:
I notice that you've changed the order in which the checkpoint files are written (previously it was `chkptbk` then `chkpt`, now the other way round).